### DISCUSSION TITLE
App-of-Apps - Ensuring Strimzi Child App Health Before Kafka App Sync

### DISCUSSION THREAD

**Discussion by fjammes** [argoproj/argo-cd#16729] [Q&A]
I'm currently working on setting up an ArgoCD deployment using the App-of-Apps pattern. In my scenario, I have a Strimzi application and a Kafka application as part of the parent application. The issue I'm facing is related to the synchronization order. The Strimzi child app needs to be healthy before the Kafka app syncs because the Kafka CRDs required by the Kafka apps are installed by the Strimzi app.

Here's a simplified version of my deployment script:

```bash
NS=argocd

argocd login --core
kubectl config set-context --current --namespace="$NS"

# Create fink app
argocd app create fink --dest-server https://kubernetes.default.svc \
    --dest-namespace "$NS" \
    --repo https://github.com/astrolabsoftware/fink-cd.git \
    --path apps --revision "$FINK_CD_WORKBRANCH" \

# Sync fink app-of-apps
argocd app sync fink

# Sync child apps
# Is there a way to remove 5 lines below and keep only the 6th
# for example using sync-waves?
argocd app sync strimzi
while ! kubectl wait --for condition=established --timeout=60s crd/kafkas.kafka.strimzi.io crd/kafkatopics.kafka.strimzi.io
do
    sleep 5
done
argocd app sync -l app.kubernetes.io/instance=fink
```

The problem is that if the Kafka app syncs first, it fails because the Kafka CRDs are missing, and they are supposed to be installed by the Strimzi app.

How can I ensure that the Strimzi child app is healthy before the Kafka app syncs when using the ArgoCD App-of-Apps pattern? Are there any specific syncPolicy configurations or strategies that can help me achieve the desired synchronization order?

Any insights or suggestions would be greatly appreciated. Thanks in advance!

**Comment by joebowbeer** [comment]
Also asked https://stackoverflow.com/q/77750481/901597

There's a health check in this article

https://oneuptime.com/blog/post/2026-02-26-deploy-strimzi-kafka-argocd/view#step-6-argocd-health-checks
